File: Interfaces/ITrImportLineService.cs
using WMS_WEBAPI.DTOs;

namespace WMS_WEBAPI.Interfaces
{
    public interface ITrImportLineService
    {
        Task<ApiResponse<IEnumerable<TrImportLineDto>>> GetAllAsync();
        Task<ApiResponse<TrImportLineDto>> GetByIdAsync(long id);
        Task<ApiResponse<IEnumerable<TrImportLineDto>>> GetByLineIdAsync(long lineId);
        Task<ApiResponse<IEnumerable<TrImportLineDto>>> GetByRouteIdAsync(long routeId);
        Task<ApiResponse<IEnumerable<TrImportLineDto>>> GetByStockCodeAsync(string stockCode);
        Task<ApiResponse<IEnumerable<TrImportLineDto>>> GetByCellCodeAsync(string cellCode);
        Task<ApiResponse<IEnumerable<TrImportLineDto>>> GetByErpOrderNoAsync(string erpOrderNo);
        Task<ApiResponse<IEnumerable<TrImportLineDto>>> GetActiveAsync();
        Task<ApiResponse<TrImportLineDto>> CreateAsync(CreateTrImportLineDto createDto);
        Task<ApiResponse<TrImportLineDto>> UpdateAsync(long id, UpdateTrImportLineDto updateDto);
        Task<ApiResponse<bool>> SoftDeleteAsync(long id);
    }
}

File: WebApi/Interfaces/ITrImportLineService.ts
import { ApiResponse } from '../Models/ApiResponse';
import { TrImportLineDto, CreateTrImportLineDto, UpdateTrImportLineDto } from '../Models/TrImportLineDtos';

export interface ITrImportLineService {
  getAll(): Promise<ApiResponse<TrImportLineDto[]>>;
  getById(id: number): Promise<ApiResponse<TrImportLineDto>>;
  getByLineId(lineId: number): Promise<ApiResponse<TrImportLineDto[]>>;
  getByRouteId(routeId: number): Promise<ApiResponse<TrImportLineDto[]>>;
  getByStockCode(stockCode: string): Promise<ApiResponse<TrImportLineDto[]>>;
  getByCellCode(cellCode: string): Promise<ApiResponse<TrImportLineDto[]>>;
  getByErpOrderNo(erpOrderNo: string): Promise<ApiResponse<TrImportLineDto[]>>;
  getActive(): Promise<ApiResponse<TrImportLineDto[]>>;
  create(createDto: CreateTrImportLineDto): Promise<ApiResponse<TrImportLineDto>>;
  update(id: number, updateDto: UpdateTrImportLineDto): Promise<ApiResponse<TrImportLineDto>>;
  softDelete(id: number): Promise<ApiResponse<boolean>>;
}